server/src/main/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderatorDriver.java

```diff
@@ -359,6 +360,7 @@ public void onSuccess(Object result)
                       if (numRemainingHandoffSegments.decrementAndGet() == 0) {
                         List<DataSegment> segments = segmentsAndCommitMetadata.getSegments();
                         log.debug("Successfully handed off [%d] segments.", segments.size());
+                        metrics.reportMaxSegmentHandoffTime(System.currentTimeMillis() - handoffStartTime);
```

Comment:
That does make sense, it would be nice to have the logs to clearly point to the problematic segments.

But I think the only option is logging this only when the `numRemainingHandoffSegments` is 0. We would also need to call out in the log that possibly only "some" of the segment ids (out of the ones in the commit metadata) are problematic ones. This should be okay as the commit metadata should not be likely to have too many segments in one batch.

Tapping into the `Object result` passed to the `onSuccess` to decide exactly which segment comes after the threshold doesn't seem viable either as the Futures seem to return weird objects (I didn't dig very deep though).

Another concern is the exact value of the threshold itself, which would have to be hard-coded.

Reply:
@abhishekagarwal87, as @kfaraz has pointed out, it seems difficult to get a good threshold since long coodinator times may affect the handoff period. 
Does 15 or maybe 30 mins seem like a good threshold to avoid too many alerts on large clusters?

Reply:
@AmatyaAvadhanula , we just need to log here for someone debugging an issue to find. We need not raise an alert here. That said, a 10 or 15 min threshold seems fine.